concurrent_map.go

```diff
@@ -64,7 +69,10 @@ func (m *ConcurrentMap) Set(key string, value interface{}) {
 	shard := m.GetShard(key)
 	shard.Lock()
 	_, ok := shard.items[key]
-	shard.items[key] = value
+	shard.items[key] = &valWithIndex{
```

Comment:
If the value is found, it should be removed first and then readded on the latest position.

Reply:
done